src/main/java/de/tum/in/www1/artemis/service/exam/ExamUserService.java

```diff
@@ -0,0 +1,123 @@
+package de.tum.in.www1.artemis.service.exam;
+
+import java.awt.*;
+import java.io.*;
+import java.util.*;
+import java.util.List;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripperByArea;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.exam.ExamUser;
+import de.tum.in.www1.artemis.repository.ExamUserRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.FileService;
+import de.tum.in.www1.artemis.web.rest.dto.ImageDTO;
+
+/**
+ * Service Implementation for managing Exam Users.
+ */
+@Service
+public class ExamUserService {
+
+    private final Logger log = LoggerFactory.getLogger(ExamUserService.class);
+
+    private final ExamUserRepository examUserRepository;
+
+    private final UserRepository userRepository;
+
+    private final FileService fileService;
+
+    public ExamUserService(FileService fileService, UserRepository userRepository, ExamUserRepository examUserRepository) {
+        this.examUserRepository = examUserRepository;
+        this.userRepository = userRepository;
+        this.fileService = fileService;
+    }
+
+    /**
+     * Extracts all images from PDF file and map each image with student registration number.
+     *
+     * @param file PDF file to be parsed
+     * @return list of ExamUserWithImageDTO
+     */
+    public List<ExamUserWithImageDTO> parsePDF(MultipartFile file) {
+
+        try (PDDocument document = PDDocument.load(file.getBytes())) {
+            ImageExtractor imageExtractor = new ImageExtractor(document);
+            imageExtractor.process();
+            List<ImageDTO> images = imageExtractor.getImages();
+            List<ExamUserWithImageDTO> studentWithImages = new ArrayList<>();
+
+            // load again as ImageExtractor closes the document
+            PDDocument document1 = PDDocument.load(file.getBytes());
```

Comment:
Maybe re-use same object? Otherwise, the try-with-resource does not work.

Reply:
I removed the step where ImageExtractor closes the document and now try-with-resource closes it.  :+1: